zio-http/src/test/scala/zio/http/RequestSpec.scala

```diff
@@ -9,7 +9,7 @@ object RequestSpec extends ZIOSpecDefault {
   def spec: Spec[TestEnvironment with Scope, Any] = suite("Result")(
     test("`#default`") {
       val body     = Body.fromString("foo")
-      val expected = Request(
+      val expected = Request.ClientRequest(
```

Comment:
There is no need for a user to know about the distinction between client request and other type of requests. So let's just make this `.apply` like before, so you can construct a request simply without implementation / performance details leaking. `ClientRequest` should not appear in the public API at all.